honeycomb-core/src/cmap2/advanced_ops.rs

```diff
@@ -0,0 +1,87 @@
+//! Advanced operations implementation
+//!
+//! This module contains code used to implement advanced operations, e.g. some non-standard,
+//! higher-level abstractions that are useful in meshing algorithms.
+
+// ------ IMPORTS
+
+use crate::{CMap2, CoordsFloat, DartIdentifier, EdgeIdentifier, Vertex2, NULL_DART_ID};
+
+// ------ CONTENT
+
+impl<T: CoordsFloat> CMap2<T> {
+    /// Split an edge into to segments.
+    ///
+    /// <div class="warning">
+    /// This implementation is 2D specific.
+    /// </div>
+    ///
+    /// This method takes all darts of an edge and rebuild connections in order to create a new
+    /// point on this edge. The position of the point default to the midway point, but it can
+    /// optionally be specified.
```

Comment:
How do you ensure that the intermediate point is on the edge. Looks like a scalar parameter like 0.5 might be better to compute a point that is really located on the edge.

Reply:
I thought that we might want to place vertices that are slightly offset from the original edges, but it's probably best to go with your idea and move vertices after if necessary.